village/src/App.js

```diff
@@ -1,27 +1,105 @@
-import React, { Component } from 'react';
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import { BrowserRouter as Router, Route } from 'react-router-dom';
 
 import './App.css';
 import SmurfForm from './components/SmurfForm';
 import Smurfs from './components/Smurfs';
+import Header from './components/Header';
+import SmurfPage from './components/SmurfPage';
 
-class App extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      smurfs: [],
-    };
+const url = 'http://localhost:3333/smurfs';
+
+function App () {
+  const [smurfs, setSmurfs] = useState(null);
+  const [editMode, setEditMode] = useState(false);
+  const [deleteMode, setDeleteMode] = useState(false);
+  
+  const getSmurfs = async () => {
+    try {
+      const res = await axios.get(url);
+      setSmurfs(res.data);
+      setDeleteMode(false);
+    } catch(e) {
+      console.log(e);
+    }
+  }
+
+  const addSmurf = async fields => {
+    const url = 'http://localhost:3333/smurfs';
```

Comment:
`url` was already defined on top of the file so probably there is no need to repeat that inside these methods.